File: src/main/java/com/group/libraryapp/domain/user/loanhistory/UserLoanHistoryRepository.java
package com.group.libraryapp.domain.user.loanhistory;

import org.springframework.data.jpa.repository.JpaRepository;

public interface UserLoanHistoryRepository extends JpaRepository<UserLoanHistory, Long> {

  JavaUserLoanHistory findByBookNameAndIsReturn(String bookName, boolean isReturn);

}

File: src/main/kotlin/com/group/libraryapp/domain/user/loanhistory/UserLoanHistoryRepository.kt
package com.group.libraryapp.domain.user.loanhistory

import com.group.libraryapp.dto.book.response.BookStatResponse
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface UserLoanHistoryRepository : JpaRepository<UserLoanHistory, Long> {

    // nullable value return
    fun findByBookNameAndStatus(bookName :String, status: UserLoanStatus) : UserLoanHistory?

    fun findAllByStatus(status: UserLoanStatus): List<UserLoanHistory>

    fun countByStatus(status: UserLoanStatus): Long

}